src/codeu/chat/client/commandline/Chat.java

```diff
@@ -39,7 +50,30 @@
   private final Stack<Panel> panels = new Stack<>();
 
   public Chat(Context context) {
+
     this.panels.push(createRootPanel(context));
+    lastLogBackup = Time.now();
+
+    try {
+      // Create a new transaction_log.txt file if needed - if not, this command does nothing
+      if(!logFile.createNewFile())
+        System.out.println("Successfully restored last logged server state.");
+
+      // Open the file as a PrintWriter and set file writing options to append
+      pw_log = new PrintWriter(new BufferedWriter(new FileWriter("data/transaction_log.txt", true)));
```

Comment:
Seems strange to use `BufferedWriter` here, since it serves the same functionality as using an `ArrayDeque`?